core/src/test/java/kafka/server/SharePartitionTest.java

```diff
@@ -118,14 +125,193 @@ public void testRecordStateForId() {
+        assertEquals(0, sharePartition.endOffset());
+        assertEquals(0, sharePartition.stateEpoch());
+        assertEquals(0, sharePartition.nextFetchOffset());
+        assertFalse(sharePartition.isReady());
+    }
+
+    @Test
+    public void testInitializeWithNullTopicsData() {
+        Persister persister = Mockito.mock(Persister.class);
+        ReadShareGroupStateResult readShareGroupStateResult = Mockito.mock(ReadShareGroupStateResult.class);
+        Mockito.when(readShareGroupStateResult.topicsData()).thenReturn(null);
+        Mockito.when(persister.readState(Mockito.any())).thenReturn(CompletableFuture.completedFuture(readShareGroupStateResult));
+        SharePartition sharePartition = sharePartition(RECORD_LOCK_DURATION_MS, MAX_DELIVERY_COUNT, persister);
+
+        assertTrue(sharePartition.cachedState().isEmpty());
+        assertEquals(0, sharePartition.startOffset());
+        assertEquals(0, sharePartition.endOffset());
+        assertEquals(0, sharePartition.stateEpoch());
+        assertEquals(0, sharePartition.nextFetchOffset());
+        assertFalse(sharePartition.isReady());
+    }
+
+    @Test
+    public void testInitializeWithEmptyTopicsData() {
+        Persister persister = Mockito.mock(Persister.class);
+        ReadShareGroupStateResult readShareGroupStateResult = Mockito.mock(ReadShareGroupStateResult.class);
+        Mockito.when(readShareGroupStateResult.topicsData()).thenReturn(Collections.emptyList());
+        Mockito.when(persister.readState(Mockito.any())).thenReturn(CompletableFuture.completedFuture(readShareGroupStateResult));
+        SharePartition sharePartition = sharePartition(RECORD_LOCK_DURATION_MS, MAX_DELIVERY_COUNT, persister);
+
+        assertTrue(sharePartition.cachedState().isEmpty());
+        assertEquals(0, sharePartition.startOffset());
+        assertEquals(0, sharePartition.endOffset());
+        assertEquals(0, sharePartition.stateEpoch());
+        assertEquals(0, sharePartition.nextFetchOffset());
+        assertFalse(sharePartition.isReady());
+    }
+
+    @Test
+    public void testInitializeWithReadException() {
+        Persister persister = Mockito.mock(Persister.class);
+        Mockito.when(persister.readState(Mockito.any())).thenReturn(FutureUtils.failedFuture(new RuntimeException("Read exception")));
+        SharePartition sharePartition = sharePartition(RECORD_LOCK_DURATION_MS, MAX_DELIVERY_COUNT, persister);
+
+        assertTrue(sharePartition.cachedState().isEmpty());
+        assertEquals(0, sharePartition.startOffset());
+        assertEquals(0, sharePartition.endOffset());
+        assertEquals(0, sharePartition.stateEpoch());
+        assertEquals(0, sharePartition.nextFetchOffset());
+        assertFalse(sharePartition.isReady());
     }
 
     @Test
     public void testAcquireSingleRecord() {
-        SharePartition sharePartition = mockSharePartition();
+        SharePartition sharePartition = sharePartition();
```

Comment:
@adixitconfluent I have changed the method name to not include `mock` as the instance returned for SharePartition is not mocked instance.